From df7495424d8937c89a36d949a0ffb74c56be460f Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Thu, 28 Oct 2010 22:35:07 +0100
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: bbe2e11a4b683ead070ad5823cdf756e1525b903
Subject: [PATCH 1817/2588] drm/i915: Do not return -1 from shrinker when
 nr_to_scan == 0

The error code is only expected during the actual pruning and not during
the first measurement (nr_to_scan == 0) pass.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit bbe2e11a4b683ead070ad5823cdf756e1525b903)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 67b2ee6..c9670e5 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -5202,7 +5202,7 @@ i915_gem_inactive_shrink(struct shrinker *shrinker,
 	int cnt;
 
 	if (!mutex_trylock(&dev->struct_mutex))
-		return nr_to_scan ? 0 : -1;
+		return 0;
 
 	/* "fast-path" to count number of available objects */
 	if (nr_to_scan == 0) {
-- 
1.7.6

